src/connection.c

```diff
@@ -353,14 +369,17 @@ static noPollConn * nopoll_tls_common_conn (noPollCtx  * ctx,char * serverAddr,c
             ParodusInfo("Connecting in Ipv6 mode\n");
             connection = nopoll_conn_tls_new6 (ctx, opts,serverAddr,serverPort,NULL,get_parodus_cfg()->webpa_path_url,NULL,NULL);
         } else {
-            ParodusInfo("Try connecting with Ipv6 mode\n");
-            connection = nopoll_conn_tls_new6 (ctx, opts,serverAddr,serverPort,NULL,get_parodus_cfg()->webpa_path_url,NULL,NULL);
-            if(connection == NULL)
-            {
-                ParodusInfo("Ipv6 connection failed. Try connecting with Ipv4 mode \n");
-                opts = createConnOpts(extra_headers, true);
-                connection = nopoll_conn_tls_new (ctx, opts,serverAddr,serverPort,NULL,get_parodus_cfg()->webpa_path_url,NULL,NULL);
-            }
+			if(FLAGS_IPV6_ONLY == (FLAGS_IPV6_IPV4 & fallback))
+			{
+				ParodusInfo("Try connecting with Ipv6 mode\n");
+				connection = nopoll_conn_tls_new6 (ctx, opts,serverAddr,serverPort,NULL,get_parodus_cfg()->webpa_path_url,NULL,NULL);
+			}
+			if(!nopoll_conn_is_ok (connection) || FLAGS_IPV4_ONLY == (FLAGS_IPV6_IPV4 & fallback))
```

Comment:
I guess we should toggle fallback if this condition becomes true 
!nopoll_conn_is_ok (connection)